CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+project(TCP-Server)
+
+cmake_minimum_required(VERSION 3.5)
+
+set(CMAKE_CXX_STANDARD 20)
+
+find_package(Valgrind)
+find_package(Seastar REQUIRED)
+
+set(RUST_TARGET_DIR ${CMAKE_CURRENT_BINARY_DIR}/rust)
+
+# Build Rust part.
+add_custom_target(
+    rust
+    COMMAND cd ${CMAKE_SOURCE_DIR}/rust &&
+            cargo build --target-dir ${RUST_TARGET_DIR}
+)
```

Comment:
It should conditionally add `--release` flag depenging on cmake mode. See here an example of that: https://github.com/scylladb/cpp-rust-driver/blob/1bcdecb3a72c62658083e90f371db1e41f0458ba/CMakeLists.txt#L239-L247
Also, the example I linked above uses ExternalProject instead of custom target - I don't remember what are the differences - why is custom_target better?